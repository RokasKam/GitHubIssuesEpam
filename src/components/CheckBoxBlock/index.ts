import { CheckBoxBlock } from "./CheckBoxBlock";
export { CheckBoxBlock };