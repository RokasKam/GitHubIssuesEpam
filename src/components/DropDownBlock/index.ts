import { DropDownBlock } from "./DropDownBlock";
export { DropDownBlock };