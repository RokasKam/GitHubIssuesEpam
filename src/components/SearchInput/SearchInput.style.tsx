import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    inputView: {
        width: "80%",
        height: 50,
        backgroundColor: "#00bf33",
        borderRadius: 30,
        marginBottom: 20
    },
    textInput: {
        height: 50,
        marginLeft: 20,
        marginRight: 20,
        color: "white",

    },
});