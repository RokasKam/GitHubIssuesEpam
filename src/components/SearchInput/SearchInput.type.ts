export type Props = {
    value?: string;
    textHandler?: (text: string) => void;
    placeholder: string;
};