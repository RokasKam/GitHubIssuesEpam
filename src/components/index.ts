export { CheckBoxBlock } from "./CheckBoxBlock";
export { DropDownBlock } from "./DropDownBlock";
export { IssuesList } from "./IssuesList";
export { ListCard } from "./ListCard";
export { SearchButton } from "./SearchButton";
export { SearchInput } from "./SearchInput";

