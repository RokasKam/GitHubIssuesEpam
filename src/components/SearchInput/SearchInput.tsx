import React from 'react'
import { View, TextInput, } from 'react-native';
import { styles } from './SearchInput.style';
import { Props } from './SearchInput.type';

export const SearchInput: React.FC<Props> = ({ value, textHandler, placeholder }) => {
    return (
        <View style={styles.inputView}>
            <TextInput
                style={styles.textInput}
                value={value}
                onChangeText={text => textHandler?.(text)}
                placeholder={placeholder}
                placeholderTextColor="white"
                autoCapitalize="none"
            />
        </View>

    )
};

