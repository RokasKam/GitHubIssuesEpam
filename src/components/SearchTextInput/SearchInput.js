import React from 'react'
import { View, TextInput, } from 'react-native';
import { styles } from './SearchInput.style';

export const SearchInput = ({value, textHandler, placeholder }) => {
    return (
        <View style={styles.inputView}>
            <TextInput
                style={styles.textInput}
                value={value}
                onChangeText={text => textHandler(text)}
                placeholder={placeholder}
                placeholderTextColor="white"
                autoCapitalize="none"
            />
        </View>

    )
};

