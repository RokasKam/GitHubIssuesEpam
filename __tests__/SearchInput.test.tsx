import React from 'react';
import renderer from 'react-test-renderer';
import { SearchInput } from '../src/components/SearchInput'

test('renders correctly', () => {
  const tree = renderer.create(<SearchInput placeholder={''} />).toJSON();
  expect(tree).toMatchSnapshot();
});