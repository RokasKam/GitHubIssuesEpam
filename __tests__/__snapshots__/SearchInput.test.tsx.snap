// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00bf33",
      "borderRadius": 30,
      "height": 50,
      "marginBottom": 20,
      "width": "80%",
    }
  }
>
  <TextInput
    autoCapitalize="none"
    onChangeText={[Function]}
    placeholder=""
    placeholderTextColor="white"
    style={
      Object {
        "color": "white",
        "height": 50,
        "marginLeft": 20,
        "marginRight": 20,
      }
    }
  />
</View>
`;
